pages/_app.js

```diff
@@ -5,7 +5,27 @@ import { medications } from "@/lib/db";
 
 export default function App({ Component, pageProps }) {
   const [medicationsList, setMedicationsList] = useState(medications);
+  const [medicationPlan, setMedicationPlan] = useState([]);
+  const [forSomeoneElse, setForSomeoneElse] = useState(false);
+  const handleForChange = (event) => {
+    setForSomeoneElse(event.target.value === "Someone else");
+  };
 
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    const formData = new FormData(event.target);
+    const frequency = [];
+    for (let [key, value] of formData.entries()) {
+      if (key.startsWith("frequency")) {
+        frequency.push(value);
+      }
+    }
+    const plan = {
+      ...Object.fromEntries(formData.entries()),
+      frequency,
+    };
```

Comment:
This whole frequency thing is very confusing. But if it works it works I guess. You could consider just working with `frequencyMornings` and the others directly in the object instead of in an array.

Reply:
Thank you for your feedback , i have already changed it